src/cedict-service/cedict.js

```diff
@@ -0,0 +1,389 @@
+   *
+   * @param {object} configuration - A JSON like configuration object.
+   */
+  static checkConfiguration (configuration) {
+    if (!configuration.storage) throw new Error('Storage tree is missing from a configuration')
+    if (!configuration.storage.stores) throw new Error('Stores data is missing from a configuration')
+    if (!configuration.storage.stores.dictionary) throw new Error('Dictionary tree is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.primaryIndex) throw new Error('A primaryIndex tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.primaryIndex.hasOwnProperty('keyPath')) throw new Error('A keyPath option of a primaryIndex tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.volatileStorage) throw new Error('A volatileStorage tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.volatileStorage.hasOwnProperty('enabled')) throw new Error('enabled option of a volatileStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.volatileStorage.hasOwnProperty('indexed')) throw new Error('indexed option of a volatileStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.permanentStorage) throw new Error('A permanentStorage tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.permanentStorage.hasOwnProperty('enabled')) throw new Error('enabled option of a permanentStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.permanentStorage.hasOwnProperty('indexed')) throw new Error('indexed option of a permanentStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.data) throw new Error('Date tree is missing from a configuration')
+    if (!configuration.data.version) throw new Error('Data version is missing from a configuration')
+    if (!configuration.data.revision) throw new Error('Data revision is missing from a configuration')
+    if (!configuration.data.recordsCount) throw new Error('Data records count is missing from a configuration')
+    if (!configuration.data.URI) throw new Error('Data URI is missing from a configuration')
+    if (!configuration.data.chunks || configuration.data.chunks.length === 0) throw new Error('Data chunks are missing from a configuration')
+  }
+
+  /**
+   * Initializes a data object.
+   *
+   * @returns {Promise<undefined> | Promise<Error>} - A promise
+   */
+  init () {
```

Comment:
@kirlat , of course each of us has his own code style preferences.
And you use fullform Promise-way for writing your code.
But from my point of view async/await style has some preferences here:
- code is much shorter (without loosing any functionality)
- code could be much easier tested (using mocks on each step)
- code is much easier to read (because it is really straightforward and you don't need to seatch then/catch for each promise, where resolve/reject are placed and so on)

For example for init () it could be overwritten this way
```
 try {
   this._storage = new CedictPermanentStorage(this._configuration.storage)
   await this._storage.connect()

   const storageData = await this._storage.getIntergrityData())
   if (this._checkOutdated(storageData)) {
    this._updateStorageData()
   } else {

    if (this._configuration.storage.stores.dictionary.volatileStorage.enabled) {
        const entries = await this._storage.stores.dictionary.getAllEntries()
        this.populateVolatileStorage(storageData.metadata, entries))
    }
   }
   return true
} catch (error) {
  console.error(error)
  return false 
}

async _updateStorageData () {
  try {
    await this._storage.destroy()
    await this._storage.connect()
    const { meta, dictionary } = await this.downloadData()
    if (this._configuration.storage.stores.dictionary.volatileStorage.enabled) {
        this.populateVolatileStorage(meta, dictionary)
    }
    await this.populatePermanentStorage(meta, dictionary)
    return true
  } catch (error) {
      ...
  }
}
```

@kirlat , what advantages do you see in this case for fullway writing code for Promise?

Reply:
Agree with your suggestion that it would be good to define our code style for promises (@irina060981 do I understand correctly that it is about whether to use a `Promise` object or an async/await combination)?

I agree that async/await code is easier to read. The reason why I used promises is that with them it is easier for me to remember that operations are asynchronous and that we are waiting for one piece of to complete to get results back while the other code is executing "in parallel". But maybe that's just my mentality. On my opinion, async/await are generally great, except for the fact that because the code with async/await is written in a way that reminds a synchronous manner strongly it is so easy to forget that it is asynchronous in fact and make a subtle mistake that will result in performance degradation. I mostly refer to cases where several async requests can run in parallel but run sequentially instead. But I think if we'll be careful such situations can be avoided.

So I think async/await is a more progressive way to write the code and it might make sense to stick to it. @balmas, what do you think?

Reply:
I agree that the async/await code is much easier to read and understand. From subsequent comments it seems that we can achieve what we need to using that structure, and if so, I think it's a good way to go.